src/coprocessor/codec/mysql/time.rs

```diff
@@ -311,6 +333,41 @@ impl Time {
         Time::new(t, tp, fsp as i8)
     }
 
+    pub fn from_duration(tz: &FixedOffset, d: &MyDuration) -> Result<Time> {
+        let local = Local::now();
+        let t = try!(ymd_hms_nanos(
+            tz,
+            local.year(),
+            local.month(),
+            local.day(),
+            0,
+            0,
+            0,
+            0
+        ));
+        let dur = Duration::nanoseconds(d.to_nanos());
+        let t = t.checked_add_signed(dur);
+        if t.is_none() {
+            Err(box_err!("parse from duration {} overflows", d))
+        } else {
+            Time::new(t.unwrap(), types::DATETIME, d.fsp as i8)
+        }
+    }
+
+    pub fn to_duration(&self) -> Result<MyDuration> {
+        if self.is_zero() {
+            return Ok(MyDuration::zero());
+        }
+        let (hour, minute, second, nanosecs) = (
+            self.time.hour() as i64,
```

Comment:
For timestamp, we would convert to `tz` in init function `from_packed_u64` @BusyJay